openbas-front/src/admin/components/simulations/simulation/reports/InjectReportResult.tsx

```diff
@@ -0,0 +1,183 @@
+  style,
+  injects,
+  initialInjectComments = [],
+  canEditComment = false,
+  onCommentSubmit = () => {},
+}) => {
+  // Standard hooks
+  const { t, fldt, tPick } = useFormatter();
+
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [injectsComments, setInjectsComments] = useState<ReportInjectComment[]>(initialInjectComments);
+  const [editInjectId, setEditInjectId] = useState<InjectResultDTO['inject_id'] | null>(null);
+
+  const findInjectCommentsByInjectId = (injectId: InjectResultDTO['inject_id']) => (injectsComments ?? []).find((c) => c.inject_id === injectId);
+
+  const handleCommentChange = (injectCommentEdited: ReportInjectComment | null, e: React.ChangeEvent<HTMLInputElement>) => {
+    if (!editInjectId) return;
+
+    const newCommentValue = e.target.value;
+
+    setInjectsComments((prevComments) => {
+      if (injectCommentEdited) {
+        return prevComments.map((comment) => (comment.inject_id === injectCommentEdited.inject_id
+          ? { ...comment, report_inject_comment: newCommentValue }
+          : comment));
+      }
+      return [
+        ...prevComments,
+        {
+          inject_id: editInjectId,
+          report_inject_comment: newCommentValue,
+        },
+      ];
+    });
+  };
+
+  useEffect(() => {
+    if (inputRef.current && editInjectId) {
+      inputRef.current.setSelectionRange(
+        findInjectCommentsByInjectId(editInjectId)?.report_inject_comment?.length || 0,
+        findInjectCommentsByInjectId(editInjectId)?.report_inject_comment?.length || 0,
+      );
+    }
+  }, [editInjectId, injectsComments]);
+
+  const saveComment = () => {
+    if (!editInjectId) return;
+    const editedComments = findInjectCommentsByInjectId(editInjectId);
+    if (editedComments) {
+      onCommentSubmit(editedComments);
+      setEditInjectId(null);
+    }
+  };
+
+  const columns = [
+    {
+      label: 'Type',
+      render: (inject: InjectResultDTO) => {
+        return inject.inject_injector_contract
+          ? <InjectorContract variant="list" label={tPick(inject.inject_injector_contract?.injector_contract_labels)} />
+          : <InjectorContract variant="list" label={t('Deleted')} deleted={true} />;
+      },
+    },
+    {
+      label: 'Title',
+      render: (inject: InjectResultDTO) => inject.inject_title,
+    },
+    {
+      label: 'Execution date',
+      render: (inject: InjectResultDTO) => <>{fldt(inject.inject_status?.tracking_sent_date)}</>,
+    },
+    {
+      label: 'Scores',
+      render: (inject: InjectResultDTO) => <AtomicTestingResult expectations={inject.inject_expectation_results} />,
+    },
+    {
+      label: 'Targets',
+      render: (inject: InjectResultDTO) => <ItemTargets targets={inject.inject_targets} />,
+    },
+    {
+      label: 'Comments',
+      render: (inject: InjectResultDTO) => {
+        const currentInjectComment = findInjectCommentsByInjectId(inject.inject_id);
+
+        return (editInjectId === inject.inject_id
+          ? <TextField
```

Comment:
This is a new behavior in the OpenBAS platform.
Are we sure this is what we want ?